### PR TITLE
Simplify controller tests

### PR DESCRIPTION
**Pull request by Mrtenz**
Rather than mocking messenger calls, I've changed the tests to use `registerActionHandler`. This has a few benefits:

- The types are inferred properly, removing the need for type casts.
- It's more readable.
- It's easier to provide some defaults, and overwrite a single action handler if needed for a specific test.

### CODE REVIEW

File: packages/snaps-controllers/src/test-utils/controller.ts [hunk 1]
```diff
@@ -212,12 +264,17 @@ export const getPersistedSnapsState = (
...
 // Mock controller messenger for Cronjob Controller
-export const getRootCronjobControllerMessenger = () =>
-  new ControllerMessenger<
+export const getRootCronjobControllerMessenger = () => {
+  const messenger = new MockControllerMessenger<
     CronjobControllerActions | AllowedActions,
     CronjobControllerEvents | AllowedEvents
   >();
 
+  jest.spyOn(messenger, 'call');
```
Comment (FrederikBolding) [inline packages/snaps-controllers/src/test-utils/controller.ts:273]: What is this for?

Comment (Mrtenz) [reply packages/snaps-controllers/src/test-utils/controller.ts:273]: That's needed for `expect(messenger.call).toHaveBeenCalledWith(...)` to work.

File: packages/snaps-controllers/src/test-utils/controller.ts [hunk 2]
```diff
@@ -15,19 +19,121 @@ import {
...
   SnapEndowments,
 } from '../snaps';
+import { MOCK_CRONJOB_PERMISSION } from './cronjob';
 import { getNodeEES, getNodeEESMessenger } from './execution-environment';
 
-export const getControllerMessenger = () =>
-  new ControllerMessenger<
+const asyncNoOp = async () => Promise.resolve();
+
+// These types are extracted from `@metamask/controllers`. Ideally they would
```
Comment (FrederikBolding) [inline packages/snaps-controllers/src/test-utils/controller.ts:27]: Can we open a PR to export these?

File: packages/snaps-controllers/src/test-utils/controller.ts [hunk 3]
```diff
@@ -15,19 +19,121 @@ import {
...
 
+  messenger.registerActionHandler(
+    'PermissionController:hasPermission',
+    (permission) => {
+      return permission !== SnapEndowments.LongRunning;
+    },
+  );
+
+  messenger.registerActionHandler(
+    'PermissionController:hasPermissions',
```
Comment (FrederikBolding) [inline packages/snaps-controllers/src/test-utils/controller.ts:85]: Is this valid? This should have multiple permissions as input?

File: packages/snaps-controllers/src/snaps/SnapController.test.ts [hunk 4]
```diff
@@ -1162,24 +1110,21 @@ describe('SnapController', () => {
...
-        resolve(true);
-      }, 200);
-    });
+    const timeoutPromise = sleep(200);
 
     expect(
       // Race the promises to check that handlerPromise does not time out
       await Promise.race([handlerPromise, timeoutPromise]),
-    ).toBe(true);
+    ).toBe('test1');
```
Comment (FrederikBolding) [inline packages/snaps-controllers/src/snaps/SnapController.test.ts:1118]: Where does `test1` come from here?

File: packages/snaps-controllers/src/snaps/SnapController.test.ts [hunk 5]
```diff
@@ -1109,14 +1058,16 @@ describe('SnapController', () => {
           id: 1,
         },
       }),
-    ).toBe(21896430);
+    ).toBe(21896426);
```
Comment (FrederikBolding) [inline packages/snaps-controllers/src/snaps/SnapController.test.ts:1061]: How is this returning a number? 🤔

File: packages/snaps-controllers/src/snaps/SnapController.test.ts [hunk 6]
```diff
@@ -2309,38 +2175,25 @@ describe('SnapController', () => {
...
-          return false;
-        });
-
       const result = await controller.installSnaps(MOCK_ORIGIN, {
         [MOCK_SNAP_ID]: { version: newVersionRange },
       });
 
-      expect(callActionMock).toHaveBeenCalledTimes(12);
-      expect(callActionMock).toHaveBeenNthCalledWith(
+      // expect(messenger.call).toHaveBeenCalledTimes(12);
```
Comment (FrederikBolding) [inline packages/snaps-controllers/src/snaps/SnapController.test.ts:2182]: ?

File: packages/snaps-controllers/src/snaps/SnapController.test.ts (hunk 4 above)
Comment (Mrtenz) [reply packages/snaps-controllers/src/snaps/SnapController.test.ts:1118]: From `DEFAULT_SNAP_BUNDLE`. It returns `method + id`.

File: packages/snaps-controllers/src/snaps/SnapController.test.ts (hunk 5 above)
Comment (Mrtenz) [reply packages/snaps-controllers/src/snaps/SnapController.test.ts:1061]: I don't know. It was doing it before too though.

File: packages/snaps-controllers/src/snaps/SnapController.test.ts (hunk 6 above)
Comment (Mrtenz) [reply packages/snaps-controllers/src/snaps/SnapController.test.ts:2182]: Leftover from testing. Will uncomment.

File: packages/snaps-controllers/src/test-utils/controller.ts (hunk 2 above)
Comment (Mrtenz) [reply packages/snaps-controllers/src/test-utils/controller.ts:27]: MetaMask/controllers#987

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**